### DISCUSSION TITLE
Github releases

### DISCUSSION THREAD

**Discussion by frgfm** [pypdfium2-team/pypdfium2#93] [Q&A] [answered]
Hi there :wave: 

I just realized that there are some pypi releases of the project but no Github release. Since I'm not sure people are aware of recent Github features, here are a few things:
- you can set CI jobs to push pypi release from your Github releases
- Github recently added automatic release note generation (comparing with your previous tag/release)
- there is one last part that is valuable I'd argue: the documentation and the pypi release don't specify the commit hash, so there is no clear way for users/dev to check the exact state of the codebase in a release :sweat_smile: Having a Github release will do this automatically :) 

One last minor advantage: at least in my case, I spent way more time on Github than pypi. GH releases show in watcher notifications and in their feed, which nice :+1: 

Happy to help if you need some help!

**Comment by mara004** [comment]
Hi,

Yes, I acknowledge it's not nice that we don't have Github releases. I'm relatively new to software development and have not worked with CI setups yet, but I'm planning to look into it. I'll have to think about doing the release process using Github Actions, but I fear there would be less manual control, so I might have to add some more assertions to the code.

The last point you mentioned, not being able to determine the commit hash for a release, is indeed a real problem that I'll have to think about.

Concerning notifications, you can also subscribe to the [RSS feed](https://pypi.org/rss/project/pypdfium2/releases.xml) on PyPI in the meantime.

Thanks for your proposal to help. If there are any concrete changes that you would like to apply, always feel free to submit a Pull Request.

**Reply by frgfm** [reply]
Hello there :) 

Thanks for the feedback!
Just to be clear, without losing any manual control, my suggestion is to:
- hold your push to pypi for a few minutes
- go to https://github.com/pypdfium2-team/pypdfium2/releases, and click on "Draft a new release"
- click on "choose a tag" and write your new tag (something like "v1.0.0") and validate by clicking the "create a new tag on publish"
- find a short title for your release and a short description (you can paste your changelog there, or generate the changelog automatically with the button)
- click on "publish release"
- then push your pypi release

You can also use Github actions to automate things but if you want manual control, there you go :+1: 

About RSS feed, yes I agree but developers that spend most of their time on github will be notified more easily about your project evolutions. As an external contributor, I cannot create a release for your project, but I can always help you set a template for your release note generation if you want!

**Reply by mara004** [reply]
Thanks for the help, I'm really appreciating this.
I could probably do it the manual way, but that would make the release process more time consuming still, so I'd prefer automation using Github Actions. Of course, I can still run the update scripts locally before triggering a release to avoid problems. I plan to look at some other repositories that use Github Actions and then start experimenting in a fork.

**Comment by mara004** [comment]
FYI, I have now started to use git tags so that users can determine the commit hash for a release.

**Reply by frgfm** [reply]
That's a great start! Thanks for this :pray:

**Reply by mara004** [reply]
I've just done a manual GitHub release for today's build. Didn't have time to look into the automated PyPI upload yet.

**Comment by mara004** [comment, marked as answer]
I have now set up a GitHub workflow to automate the release process. See the commits d42625067a1bf46f62856dbe5f0451193d0fb3ec,  1926ffc7d905caa2769da5ce183b44ea89c60afd and 8a96164c234ce72357a1f7dbabe7b81e7fa5adc9. I was able to test most steps in a fork and hope that the final part works as well.

**Reply by mara004** [reply]
The first automated release was made today.

**Comment by mara004** [comment]
Today I changed the workflows so that a release is now triggered automatically every monday at 12 o'clock UTC (that is, 6 hours after the schedule of pdfium-binaries). No developer interaction is needed for this, so I don't have to spend time on a repetitive task, while the pdfium binaries pypdfium2 ships with still get updated on a regular basis.
